addon/components/swiper-container.js

```diff
@@ -101,6 +101,11 @@ export default Component.extend({
       options.autoplayDisableOnInteraction = this.get('autoplayDisableOnInteraction');
     }
 
+    if (this.get('watchSlidesVisibility')) {
+      options.watchSlidesProgress = true;
```

Comment:
I see your point, but still.. Just pass them as they are in split options. One thing less to document in this project. The less logic, the better.